magenta/music/musicxml_parser_test.py

```diff
@@ -0,0 +1,190 @@
+    for musicxml_tempo, sequence_tempo in zip(
+        musicxml_tempos, sequence_proto.tempos):
+      self.assertAlmostEqual(musicxml_tempo.qpm, sequence_tempo.qpm)
+      self.assertAlmostEqual(musicxml_tempo.time_position,
+                             sequence_tempo.time)
+
+    # Test parts/instruments.
+    seq_instruments = defaultdict(lambda: defaultdict(list))
+    for seq_note in sequence_proto.notes:
+      seq_instruments[
+          (seq_note.instrument, seq_note.program)]['notes'].append(seq_note)
+
+    sorted_seq_instrument_keys = sorted(
+        seq_instruments.keys(),
+        key=lambda (instrument_id, program_id): (instrument_id, program_id))
+
+    self.assertEqual(len(musicxml.parts), len(seq_instruments))
+    for musicxml_part, seq_instrument_key in zip(
+        musicxml.parts, sorted_seq_instrument_keys):
+
+      seq_instrument_notes = seq_instruments[seq_instrument_key]['notes']
+      musicxml_notes = []
+      for musicxml_measure in musicxml_part.measures:
+        for musicxml_note in musicxml_measure.notes:
+          if not musicxml_note.is_rest:
+            musicxml_notes.append(musicxml_note)
+
+      self.assertEqual(len(musicxml_notes), len(seq_instrument_notes))
+      for musicxml_note, sequence_note in zip(musicxml_notes,
+                                              seq_instrument_notes):
+        self.assertEqual(musicxml_note.pitch[1], sequence_note.pitch)
+        self.assertEqual(musicxml_note.velocity, sequence_note.velocity)
+        self.assertAlmostEqual(musicxml_note.note_duration.time_position,
+                               sequence_note.start_time)
+        self.assertAlmostEqual(musicxml_note.note_duration.time_position \
+                               + musicxml_note.note_duration.seconds,
+                               sequence_note.end_time)
+        # Check that the duration specified in the MusicXML and the
+        # duration float match to within +/- 1 (delta = 1)
+        # Delta is used because duration in MusicXML is always an integer
+        # For example, a 3:2 half note might have a durationfloat of 341.333
+        # but would have the 1/3 distributed in the MusicXML as
+        # 341.0, 341.0, 342.0.
+        # Check that (3 * 341.333) = (341 + 341 + 342) is true by checking
+        # that 341.0 and 342.0 are +/- 1 of 341.333
+        self.assertAlmostEqual(musicxml_note.note_duration.duration, \
+                               musicxml_note.state.divisions * 4 \
+                               * musicxml_note.note_duration.durationfloat(), \
+                               delta=1)
+
+  def checkmusicxmltosequence(self, filename):
+    """Test the translation from MusicXML to Sequence proto."""
+    source_musicxml = MusicXMLDocument(filename)
+    sequence_proto = musicxml_to_sequence_proto(source_musicxml)
+    self.checkmusicxmlandsequence(source_musicxml, sequence_proto)
+
+  #def testsimplemusicxmltosequence(self):
```

Comment:
Because I was debugging the rhythm test and forgot to uncomment them :). Fixed in commit 4ebfb5a